tests/components/totalconnect/common.py

```diff
@@ -0,0 +1,136 @@
+    "UserID": "1234567",
+    "Username": "username",
+    "UserFeatureList": "Master=0,User Administration=0,Configuration Administration=0",
+}
+
+RESPONSE_AUTHENTICATE = {
+    "ResultCode": 0,
+    "SessionID": 1,
+    "Locations": LOCATIONS,
+    "ModuleFlags": MODULE_FLAGS,
+    "UserInfo": USER,
+}
+
+PARTITION_DISARMED = {
+    "PartitionID": "1",
+    "ArmingState": TotalConnectClient.TotalConnectLocation.DISARMED,
+}
+
+PARTITION_ARMED_STAY = {
+    "PartitionID": "1",
+    "ArmingState": TotalConnectClient.TotalConnectLocation.ARMED_STAY,
+}
+
+PARTITION_ARMED_AWAY = {
+    "PartitionID": "1",
+    "ArmingState": TotalConnectClient.TotalConnectLocation.ARMED_AWAY,
+}
+
+PARTITION_INFO_DISARMED = {}
+PARTITION_INFO_DISARMED[0] = PARTITION_DISARMED
+
+PARTITION_INFO_ARMED_STAY = {}
+PARTITION_INFO_ARMED_STAY[0] = PARTITION_ARMED_STAY
+
+PARTITION_INFO_ARMED_AWAY = {}
+PARTITION_INFO_ARMED_AWAY[0] = PARTITION_ARMED_AWAY
+
+
+PARTITIONS_DISARMED = {"PartitionInfo": PARTITION_INFO_DISARMED}
+PARTITIONS_ARMED_STAY = {"PartitionInfo": PARTITION_INFO_ARMED_STAY}
+PARTITIONS_ARMED_AWAY = {"PartitionInfo": PARTITION_INFO_ARMED_AWAY}
+
+ZONE_NORMAL = {
+    "ZoneID": "1",
+    "ZoneDescription": "Normal",
+    "ZoneStatus": TotalConnectClient.ZONE_STATUS_NORMAL,
+    "PartitionId": "1",
+}
+
+ZONE_INFO = []
+ZONE_INFO.append(ZONE_NORMAL)
+ZONES = {"ZoneInfo": ZONE_INFO}
+
+METADATA_DISARMED = {
+    "Partitions": PARTITIONS_DISARMED,
+    "Zones": ZONES,
+    "PromptForImportSecuritySettings": False,
+    "IsInACLoss": False,
+    "IsCoverTampered": False,
+    "Bell1SupervisionFailure": False,
+    "Bell2SupervisionFailure": False,
+    "IsInLowBattery": False,
+}
+
+METADATA_ARMED_STAY = METADATA_DISARMED.copy()
+METADATA_ARMED_STAY["Partitions"] = PARTITIONS_ARMED_STAY
+
+METADATA_ARMED_AWAY = METADATA_DISARMED.copy()
+METADATA_ARMED_AWAY["Partitions"] = PARTITIONS_ARMED_AWAY
+
+RESPONSE_DISARMED = {"ResultCode": 0, "PanelMetadataAndStatus": METADATA_DISARMED}
+RESPONSE_ARMED_STAY = {"ResultCode": 0, "PanelMetadataAndStatus": METADATA_ARMED_STAY}
+RESPONSE_ARMED_AWAY = {"ResultCode": 0, "PanelMetadataAndStatus": METADATA_ARMED_AWAY}
+
+RESPONSE_ARM_SUCCESS = {"ResultCode": TotalConnectClient.TotalConnectClient.ARM_SUCCESS}
+RESPONSE_ARM_FAILURE = {
+    "ResultCode": TotalConnectClient.TotalConnectClient.COMMAND_FAILED
+}
+RESPONSE_DISARM_SUCCESS = {
+    "ResultCode": TotalConnectClient.TotalConnectClient.DISARM_SUCCESS
+}
+RESPONSE_DISARM_FAILURE = {
+    "ResultCode": TotalConnectClient.TotalConnectClient.COMMAND_FAILED,
+    "ResultData": "Command Failed",
+}
+
+
+async def setup_platform(hass, platform):
+    """Set up the TotalConnect platform."""
+    # first set up a config entry and add it to hass
+    mock_entry = MockConfigEntry(
+        domain=DOMAIN,
+        data={CONF_USERNAME: "user@email.com", CONF_PASSWORD: "password"},
+    )
+    mock_entry.add_to_hass(hass)
+
+    RESPONSES = [RESPONSE_AUTHENTICATE, RESPONSE_DISARMED]
```

Comment:
`RESPONSES` can be a local variable, not a constant.

Reply:
done